packages/react-components/src/gallery/dominantSpeaker.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { VideoGalleryRemoteParticipant } from '../types';
+
+/**
+ * Calculates the participants that should be renderd basis on the list of dominant speakers and currently renderd participants in a call.
+ * @param participants - Array containing all participants of a call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param dominantSpeakers - An array containing the userId of dominant speakers in a call in the order of their dominance. 0th index is the most dominant, 1st is the second most etc
+ * @param visibleParticipants - Array containing currently rendered (visible) participants in the call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param MAX_TILES - Maximum number of tiles to calculate.
+ * @returns VideoGalleryRemoteParticipant[] {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ */
+export const smartDominantSpeakerParticipants = (
+  participants: VideoGalleryRemoteParticipant[] = [],
+  dominantSpeakers: Array<string> = [],
+  visibleParticipants: VideoGalleryRemoteParticipant[] = [],
+  MAX_TILES = 4
+): VideoGalleryRemoteParticipant[] => {
+  if (!participants) return [];
+
+  // Only use the Max allowed dominant speakers.
+  const dominantSpeakerIds = dominantSpeakers.slice(0, MAX_TILES);
+
+  // Don't apply any logic if total number of video streams is less than Max video streams.
+  if (participants.length <= MAX_TILES) return participants;
```

Comment:
nit: This should be before the logic in line 22 (we don't even need the dominant speakers, so why clip it?)

Reply:
Just a coding style I prefer. Defining important variables up top.